### PR TITLE
New category icons for <mkt-nav> (bug 1148164, 1148166)

### PR DESCRIPTION
**Pull request by mstriemer**
screenshot 2015-04-07 15 43 22
screenshot 2015-04-07 16 18 56
screenshot 2015-04-07 16 17 32screenshot 2015-04-07 16 17 40
desktop-categories-hover mov

### CODE REVIEW

File: src/media/css/category-icons.styl [hunk 1]
```diff
@@ -6,6 +6,53 @@
 @import 'lib';
 
 $cat-icon-size-default = 28px;
+$categories = {
```
Comment (ngokevin) [inline src/media/css/category-icons.styl]: nice mathing

File: src/media/css/elements--nav.styl [hunk 2]
```diff
@@ -276,6 +277,78 @@ mkt-nav-child {
     display: none;
 }
 
+.category-list {
```
Comment (ngokevin) [inline src/media/css/elements--nav.styl]: Should probably go into its own stylesheet and not the main nav stylesheet

File: src/media/css/categories.styl [hunk 3]
```diff
@@ -107,3 +108,61 @@
...
+                category-sprite(2);
+                content: '';
+                display: inline-block;
+                left: 0;
+                margin: ($cat-icon-size-default / -2) ($cat-icon-size-default / 2);
+                opacity: 1;
+                position: absolute;
+            }
+            &:before {
+                background: #4CB1FF;
```
Comment (ngokevin) [inline src/media/css/categories.styl]: this might be ```$action-positive```

File: src/media/css/categories.styl [hunk 4]
```diff
@@ -107,3 +108,61 @@
...
+                border-radius: 50%;
+                opacity: 1;
+                content: '';
```
Comment (ngokevin) [inline src/media/css/categories.styl]: abc

File: src/media/css/category-icons.styl (hunk 1 above)
Comment (ngokevin) [inline src/media/css/category-icons.styl]: maybe could belong in lib/categories.styl?

**Comment by ngokevin** [comment]
Looks good, I would maybe add a very small transition on the background color change (like 100-200ms). r+wc

**Comment by mstriemer** [comment]
I played around with a transition and it doesn't end up working out very well. Since the border is a different colour it transitions faster than the background and looks funny. Removing the border on hover means sliding your mouse across an element immediately removes the border and then it reappears. 

The icon also transitions immediately which is partly resolved by overlaying the blue and white icons and transitioning their opacity but I can't quite get them to line up exactly and the longer transition time lets you notice that they're slightly off.

**Comment by ngokevin** [comment]
Okay, let's just forget the transitions. It was a bit silly to propose since the icons obviously can't transition.

**Comment by mstriemer** [comment]
I moved the mixins into `lib/mixins.styl` since importing a file with styles in it appears to be a bad idea. I also made the categories style (white on blue) the default for a `mkt-nav-child` but left the icons in the `header--categories.styl` file.

**Pull request merged**